### PR TITLE
feat(manager): implement "commit" method in class "Dataset"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2pz17n5 Dataset实现commit方法](https://app.clickup.com/t/2pz17n5)

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/sheets.py [hunk 1]
```diff
@@ -6,12 +6,14 @@
 """The implementation of the Sheets."""
 
 from functools import partial
-from typing import TYPE_CHECKING, Any, Dict, Iterator, MutableMapping
+from typing import TYPE_CHECKING, Any, Dict, Iterator, MutableMapping, Tuple
 
 import pyarrow as pa
+from tqdm import tqdm
```
Comment (linjiX) [inline graviti/manager/sheets.py:12]: `tqdm.auto`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 2]
```diff
@@ -226,6 +226,26 @@ def edit(
...
+            description: The commit description.
+            jobs: The number of the max workers in multi-thread upload, the default is 8.
+            quiet: Set to True to stop showing the upload process bar.
+
+        """
+        head = self.HEAD
+        draft = self.drafts.create(
+            title, description, branch=head.name  # type: ignore[attr-defined]
+        )
+        draft.operations = self.HEAD.operations
```
Comment (linjiX) [inline graviti/manager/dataset.py:245]: `head.operation`?

File: graviti/manager/dataset.py [hunk 3]
```diff
@@ -226,6 +226,26 @@ def edit(
...
+            description: The commit description.
+            jobs: The number of the max workers in multi-thread upload, the default is 8.
+            quiet: Set to True to stop showing the upload process bar.
+
+        """
+        head = self.HEAD
+        draft = self.drafts.create(
+            title, description, branch=head.name  # type: ignore[attr-defined]
+        )
+        draft.operations = head.operations
```
Comment (linjiX) [inline graviti/manager/dataset.py:245]: does this line make sense?

File: graviti/manager/dataset.py [hunk 4]
```diff
@@ -226,6 +226,26 @@ def edit(
...
+        Arguments:
+            title: The commit title.
+            description: The commit description.
+            jobs: The number of the max workers in multi-thread upload, the default is 8.
+            quiet: Set to True to stop showing the upload process bar.
+
+        """
+        head = self.HEAD
+        draft = self.drafts.create(
+            title, description, branch=head.name  # type: ignore[attr-defined]
```
Comment (linjiX) [inline graviti/manager/dataset.py:243]: what if the head is not a branch?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 5]
```diff
@@ -226,6 +226,34 @@ def edit(
...
+        """Create, upload and commit the draft to push the local dataset to Graviti.
+
+        Arguments:
+            title: The commit title.
+            description: The commit description.
+            jobs: The number of the max workers in multi-thread upload, the default is 8.
+            quiet: Set to True to stop showing the upload process bar.
+
+        Raises:
+            StatusError: When the HEAD of the dataset is a COMMIT or a TAG.
```
Comment (linjiX) [inline graviti/manager/dataset.py:241]: `is commit or tag` or `not branch` which one is better?

**Review by linjiX — APPROVED** [review]

File: graviti/manager/dataset.py [hunk 6]
```diff
@@ -226,6 +226,34 @@ def edit(
...
+
+        """
+        head = self.HEAD
+        if not isinstance(head, Branch):
+            raise StatusError(
+                "It is not allowed to commit a dataset whose HEAD is not a branch. "
+                "Please checkout a branch first"
+            )
+        if not head.operations:
+            raise StatusError("It is not allowed to commit a dataset without any modifications")
```
Comment (linjiX) [inline graviti/manager/dataset.py:252]: better to add a blank line after raise

**Pull request merged**